### ISSUE TITLE
Plugin: HDF5 files for multivolumes

### ISSUE THREAD

**Issue by clbarnes** [imageio/imageio#645]
Requires the h5py library. Also loads `.mat` files >= v7.3.

**Comment by FirefoxMetzger** [comment]
Just to make sure I got this correctly, you are proposing a new plugin that has `h5py` as a backend? This would then add HDF5 support and .mat support (among others).

(there is already a way to read/write HDF5 via the `itk` plugin, if you happen to have that backend)

**Comment by clbarnes** [comment]
Yes, a plugin for HDF5 files (to take `.hdf5`, `.h5`, `.hdf` extensions, which are all bafflingly present in the wild) using h5py, although it could be factored around ITK instead (and/or PIL? which also seems to have some HDF5iness about it). My sense is that ITK is quite a heavy dependency for something like this, though, and PIL probably represents an atypical use case.

**Comment by FirefoxMetzger** [comment]
I am not at all against adding h5py as a backend. I think it would be a very cool thing actually 🚀 .

Mainly asked to give this issue the correct tags.
